source/uri/Endpoint.ooc

```diff
@@ -0,0 +1,47 @@
+Endpoint: class {
+	_host: VectorList<Text>
+	_port: Int
+	host: VectorList<Text> {
+		get {
+			result := VectorList<Text> new()
+			for (i in 0 .. this _host count)
+				result add(this _host[i] take())
+			result
+		}
+	}
+	port ::= this _port
+
+	init: func (=_host, =_port)
+	free: override func {
+		for (i in 0 .. this _host count)
+			this _host[i] free(Owner Receiver)
+		this _host free()
+		super()
+	}
+	toText: func -> Text {
+		result := Text new()
+		for (i in 0 .. this _host count)
+			result += i == 0 ? this _host[i] take() : t"." + this _host[i] take()
+		if (this _port != 0) {
+			portString := this _port toString()
+			result += t":" + Text new(portString)
+			portString free()
+		}
+		result
+	}
+	parse: static func(data: Text) -> This {
+		result: This
+		d := data take()
+		if (!d isEmpty) {
+			splitted := d split(t":")
+			port := splitted count > 1 ? splitted remove() toInt() : 0
+			host := splitted remove() split(t".")
+			for (i in 0 .. host count)
+				host[i] = host[i] claim()
```

Comment:
If I don't use `claim` the `Text` object owner was `Unknown` and then if `take` is called the results owner is still `Unknown` and then when you free with `Owner Receiver` the object is not freed. Maybe I have misunderstood something.